src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/IncludeRewriter.cs

```diff
@@ -63,159 +63,103 @@ public override Expression VisitSqlRoot(SqlRootExpression expression, object con
-            public static IList<TableExpression> Sort(IEnumerable<TableExpression> includeExpressions)
+            if (expressions.Count == 1)
             {
-                var graph = new IncludeExpressionDependencyGraph(includeExpressions);
-                var sortedExpressions = new List<TableExpression>();
+                return expressions;
+            }
 
-                while (graph.NodesWithoutIncomingEdges.Any())
-                {
-                    // Remove a node without incoming edges and add to the sorted list
-                    var v = graph.NodesWithoutIncomingEdges.First();
-                    sortedExpressions.Add(v);
+            var graph = new IncludeIterateExpressionDependencyGraph(expressions);
+            var sortedExpressions = new List<TableExpression>();
 
-                    graph.RemoveNodeAndAllOutgoingEdges(v);
-                }
+            while (graph.NodesWithoutIncomingEdges.Any())
+            {
+                // Remove a node without incoming edges and add to the sorted list
+                var v = graph.NodesWithoutIncomingEdges.First();
+                sortedExpressions.Add(v);
 
-                return sortedExpressions;
+                graph.RemoveNodeAndAllOutgoingEdges(v);
             }
 
-            // Dependency graph of parameters so that parameter b depends on a means that b includes from a, therefore a should appear before b.
-            private class IncludeExpressionDependencyGraph
+            // If there are edges, then the graph contains a cycle
+            if (graph.OutgoingEdges.Any())
             {
-                // private static readonly IncludeExpressionComparer Comparer = new IncludeExpressionComparer();
-                public IncludeExpressionDependencyGraph(IEnumerable<TableExpression> includeExpressions)
-                {
-                    OutgoingEdges = new Dictionary<TableExpression, IList<TableExpression>>();
-                    IncomingEdgesCount = new Dictionary<TableExpression, int>();
-
-                    // Add graph nodes (parameters) and edges (dependencies)
-                    foreach (var v in includeExpressions)
-                    {
-                        OutgoingEdges.Add(v, new List<TableExpression>());
-                        IncomingEdgesCount.TryAdd(v, 0);
-
-                        foreach (var u in includeExpressions)
-                        {
-                            // if (v != u && Comparer.Compare(v, u) < 0)
-                            if (v != u && IsDependencyEdge(v, u))
-                            {
-                                IncomingEdgesCount.TryAdd(u, 0);
-                                OutgoingEdges[v].Add(u);
-                                IncomingEdgesCount[u]++;
-                            }
-                        }
-                    }
-                }
+                throw new InvalidSearchOperationException("Cyclic include iterate queries are not allowed");
```

Comment:
"not supported". And please put the string in a resx file.

Reply:
Oops.. Saw that after I merged. 
The reason I didn;t add the string to resx is that I wanted to be consistent with how all the Rewriters throw InvalidSearchOperationException.